content/doc/book/system-administration/authenticating-scripted-clients.adoc

```diff
@@ -0,0 +1,378 @@
+
+        // Then provide the right credentials
+        client.getCredentialsProvider().setCredentials(
+                new AuthScope(AuthScope.ANY_HOST, AuthScope.ANY_PORT),
+                new UsernamePasswordCredentials(username, password));
+
+        // Generate BASIC scheme object and stick it to the execution context
+        BasicScheme basicAuth = new BasicScheme();
+        BasicHttpContext context = new BasicHttpContext();
+        context.setAttribute("preemptive-auth", basicAuth);
+
+        // Add as the first (because of the zero) request interceptor
+        // It intercepts the request and initializes the authentication scheme
+        client.addRequestInterceptor(new PreemptiveAuth(), 0);
+
+        // You get request that will start the build
+        String getUrl = jenkinsUrl + "/job/" + jobName + "/build?token=" + buildToken;
+        HttpGet get = new HttpGet(getUrl);
+
+        try {
+            // Execute your request with the given context
+            HttpResponse response = client.execute(get, context);
+            HttpEntity entity = response.getEntity();
+            EntityUtils.consume(entity);
+        }
+        catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Preemptive authentication interceptor
+     *
+     */
+    static class PreemptiveAuth implements HttpRequestInterceptor {
+
+        /*
+         * (non-Javadoc)
+         *
+         * @see org.apache.http.HttpRequestInterceptor#process(org.apache.http.HttpRequest,
+         * org.apache.http.protocol.HttpContext)
+         */
+        public void process(HttpRequest request, HttpContext context)
+            throws HttpException, IOException {
+            // Get the AuthState
+            AuthState authState = (AuthState) context.getAttribute(ClientContext.TARGET_AUTH_STATE);
+
+            // If no auth scheme available yet, try to initialize it preemptively
+            if (authState.getAuthScheme() == null) {
+                AuthScheme authScheme = (AuthScheme) context.getAttribute("preemptive-auth");
+                CredentialsProvider credsProvider = (CredentialsProvider) context
+                        .getAttribute(ClientContext.CREDS_PROVIDER);
+                HttpHost targetHost = (HttpHost) context.getAttribute(ExecutionContext.HTTP_TARGET_HOST);
+                if (authScheme != null) {
+                    Credentials creds = credsProvider.getCredentials(
+                        new AuthScope(targetHost.getHostName(), targetHost.getPort())
+                    );
+                    if (creds == null) {
+                        throw new HttpException("No credentials for preemptive authentication");
+                    }
+                    authState.setAuthScheme(authScheme);
+                    authState.setCredentials(creds);
+                }
+            }
+        }
+
+    }
+}
+----
```

Comment:
The httpclient library was last released over 8 years ago, has [no available online documentation](https://stackoverflow.com/questions/29295831/where-is-the-documentation-for-httpclient-4-1-x), and has been superseded by the later example in this page.
```suggestion
```